### PR TITLE
upgrade to 6.4.0.Final

### PR DESCRIPTION
**Pull request by Fabryprog**
Upgrade to 6.4.0.Final

Replace StatefulSession into StatelessSession
Load resource from classpath

### CODE REVIEW

**Comment by gaetanoserio** [comment]
+1

**Comment by Fabryprog** [comment]
+1

File: src/main/java/pl/maciejwalkowiak/drools/DroolsSessionImpl.java [hunk 1]
```diff
@@ -13,31 +13,31 @@
 public class DroolsSessionImpl implements DroolsSession {
     private static final Logger LOG = LoggerFactory.getLogger(DroolsSessionImpl.class);
 
-    private StatefulSession statefulSession;
-
-    public DroolsSessionImpl(StatefulSession statefulSession) {
-        this.statefulSession = statefulSession;
-    }
-
+//    private StatefulKnowledgeSession statefulSession;
```
Comment (maciejwalkowiak) [inline src/main/java/pl/maciejwalkowiak/drools/DroolsSessionImpl.java:16]: Clean up unused/commented out code please.

File: src/test/java/pl/maciejwalkowiak/drools/BeforeMethodBasedTest.java [hunk 2]
```diff
@@ -22,9 +23,10 @@ public void initDrools() throws Exception {
     public void should_set_discount() {
         Purchase purchase = new Purchase(new Customer(17));
 
-        session.insert(purchase);
-        session.fireAllRules();
+        session.execute(purchase);
 
+        System.out.println("isDiscount?: " + purchase.getTicket().isDiscount());
```
Comment (maciejwalkowiak) [inline src/test/java/pl/maciejwalkowiak/drools/BeforeMethodBasedTest.java:28]: There is no need to print out anything in unit tests, and if there is for some reason - use logger.

**Comment by ashgh** [comment]
When will 6.4.0 support be merged?

**Comment by Fabryprog** [comment]
@maciejwalkowiak I forgot this PR.... Do I have to review the code?

**Comment by maciejwalkowiak** [comment]
First of all I am sorry for long delay.
I haven't been working with Drools for quite a while and this repository felt pretty abandoned.

Great job @Fabryprog. Thanks a lot for contribution. Before I can merge it I would like to know the reasons you switched from Stateful to Stateless session?

**Comment by Fabryprog** [comment]
@maciejwalkowiak i don't remember! Perhaps it would be good both be present

**Comment by maciejwalkowiak** [comment]
Ok. First I will bring back Stateful session and then will see how to easily add both.

**Comment by yangyanqing** [comment]
> Great job @Fabryprog. Thanks a lot for contribution. Before I can merge it I would like to know the reasons you switched from Stateful to Stateless session?

Too mary types in 6.1.0 Final did not exist in 6.4.0 Final, that may be the reason.

Thank you @Fabryprog & @maciejwalkowiak

**Comment by yangyanqing** [comment]
> Great job @Fabryprog. Thanks a lot for contribution. Before I can merge it I would like to know the reasons you switched from Stateful to Stateless session?

Just now，I thought Stateful was better than Stateless,  because more than two entities were needed in UT.

I am looking forward to the release of the new version. :)

**Comment by maciejwalkowiak** [comment]
Unfortunately due to the lack of time I am not able to continue developing and maintaining `junit-drools` project. Thank you for your contributions and I hope you will find one of the forks still maintained and helpful for you!

**Pull request closed without merging**